webboot/webboot_test.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/hex"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+type test struct {
+	name         string
+	linkOrName   string
+	md5link      string
+	expectedLink string
+	expectedName string
+}
+
+func TestParseArg(t *testing.T) {
+	tests := []test{
+		{name: "Debian", linkOrName: "https://cdimage.debian.org/debian-cd/current/amd64/iso-cd/debian-10.0.0-amd64-netinst.iso", expectedLink: "https://cdimage.debian.org/debian-cd/current/amd64/iso-cd/debian-10.0.0-amd64-netinst.iso", expectedName: "debian-10.0.0-amd64-netinst.iso"},
+		{name: "TinyCore", linkOrName: "tinycore", expectedLink: "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso", expectedName: "tinycore"},
+		{name: "Ubuntu", linkOrName: "http://releases.ubuntu.com/18.04.2/ubuntu-18.04.2-desktop-amd64.iso", expectedLink: "http://releases.ubuntu.com/18.04.2/ubuntu-18.04.2-desktop-amd64.iso", expectedName: "ubuntu-18.04.2-desktop-amd64.iso"},
+	}
+
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			link, filename, err := parseArg(v.linkOrName)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if strings.Compare(link, v.expectedLink) != 0 {
+				t.Errorf("Looking up URL for %v: got %v, want %v", v.linkOrName, link, v.expectedLink)
+			}
+			if strings.Compare(filename, v.expectedName) != 0 {
+				t.Errorf("Looking up the file name for %v: got %v, want %v", v.linkOrName, filename, v.expectedName)
+			}
+		})
+	}
+}
+
+func TestName(t *testing.T) {
+	tests := []test{
+		{name: "TinyCore", linkOrName: "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso", expectedName: "CorePure64-10.1.iso"},
+		{name: "ArchLinux", linkOrName: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/archlinux-2019.08.01-x86_64.iso", expectedName: "archlinux-2019.08.01-x86_64.iso"},
+		{name: "Ubuntu", linkOrName: "http://releases.ubuntu.com/18.04.2/ubuntu-18.04.2-desktop-amd64.iso", expectedName: "ubuntu-18.04.2-desktop-amd64.iso"},
+		{name: "HtmlFile", linkOrName: "http://releases.ubuntu.com", expectedName: "index.html"},
+	}
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			filename, err := name(v.linkOrName)
+			if err != nil {
+				t.Fatalf("Looking up  %v: got %v, want nil", v.linkOrName, err)
```

Comment:
Shouldn't this be
Looking up the file name for %v: got %v, want %v
like it is above?

Reply:
Yeah, it is. I see what you are saying. I changed the error message so it says something along the lines of "Looking up the file name for %v: got %v, want %v"